File: src/008. Largest product in a series/008.rb
num_str = File.read('008.txt')
digits = num_str.chars.map(&:to_i)
max_product = 0
0.upto(digits.length - 12) do |i|
  product = digits[i, 13].inject(:*)
  max_product = product if max_product < product
end

puts max_product

File: src/008. Largest product in a series/008.py
from functools import reduce
from operator import mul

with open("008.txt") as f:
    digits = [int(d) for d in f.read()]
    max_product = 0
    for i in range(0, len(digits) - 12):
        product = reduce(mul, digits[i:i + 13])
        if product > max_product:
            max_product = product
    print(max_product)
